### PR TITLE
fix Coq 8.10 warnings and build examples on CI

### LINKED ISSUE #17: Build examples on CI
@palmskog how can we make CI run `make examples-plain` and not just `make`? Would be nice to make sure that these examples keep working.

### PR DESCRIPTION
**Pull request by RalfJung**
This fixes the warnings that are shown when building with Coq 8.10.

Fixes https://github.com/coq-community/autosubst/issues/17.

### CODE REVIEW

**Comment by RalfJung** [comment]
`Declare Scope` does not work with Coq 8.9. Any objections to raising the minimum Coq version to 8.10?

**Comment by RalfJung** [comment]
@co-dan I assume you have notifications enabled for this repo so I don't need to ping you? Just making sure.^^

**Comment by RalfJung** [comment]
@palmskog I added a package, but this does not seem enough? I cannot see any trace of the examples being built in the log.

File: coq-autosubst-examples.opam [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+homepage: "https://github.com/coq-community/autosubst"
+dev-repo: "git+https://github.com/coq-community/autosubst.git"
+bug-reports: "https://github.com/coq-community/autosubst/issues"
+license: "MIT"
+
+synopsis: "Autosubst Examples (CI only)"
+
+build: [make "-j%{jobs}%" "examples-plain"]
+install: []
+depends: ["coq-autosubst"]
```
Comment (RalfJung) [inline coq-autosubst-examples.opam]: Actually this will just build the lib due to the makefile dependency, so this dependency here makes little sense...

**Comment by RalfJung** [comment]
All right, now it seems to build the (non-ssreflect) examples as well.

**Comment by RalfJung** [comment]
@co-dan what do you think about this PR?

**Comment by co-dan** [comment]
I don't really know much about the CI, so I do not understand this, but I have full trust in you :)

**Comment by RalfJung** [comment]
@co-dan and are you okay with dropping support for Coq 8.9?

**Comment by co-dan** [comment]
Yes, it's fine with me.

**Pull request merged**